plugins/modules/identity/ipa/ipa_otptoken.py

```diff
@@ -0,0 +1,530 @@
+    - In the format C(YYYYMMddHHmmss).
+    - For example, C(20200121182022) will allow the token to be used until 21 January 2020 at 18:20:22.
+    type: str
+  vendor:
+    description: Token vendor name (informational only).
+    type: str
+  model:
+    description: Token model (informational only).
+    type: str
+  serial:
+    description: Token serial (informational only).
+    type: str
+  state:
+    description: State to ensure.
+    choices: ['present', 'absent']
+    default: 'present'
+    type: str
+  algorithm:
+    description:
+    - Token hash algorithm.
+    - "B(Note:) Cannot be modified after OTP is created."
+    choices: ['sha1', 'sha256', 'sha384', 'sha512']
+    type: str
+  digits:
+    description:
+    - Number of digits each token code will have.
+    - "B(Note:) Cannot be modified after OTP is created."
+    choices: [ 6, 8 ]
+    type: int
+  offset:
+    description:
+    - TOTP token / IPA server time difference.
+    - "B(Note:) Cannot be modified after OTP is created."
+    type: int
+  interval:
+    description:
+    - Length of TOTP token code validity in seconds.
+    - "B(Note:) Cannot be modified after OTP is created."
+    type: int
+  counter:
+    description:
+    - Initial counter for the HOTP token.
+    - "B(Note:) Cannot be modified after OTP is created."
+    type: int
+extends_documentation_fragment:
+- community.general.ipa.documentation
+
+requirements:
+- base64
+'''
+
+EXAMPLES = r'''
+- name: Create a totp for pinky, allowing the IPA server to generate using defaults
+  community.general.ipa_otptoken:
+    uniqueid: Token123
+    otptype: totp
+    owner: pinky
+    ipa_host: ipa.example.com
+    ipa_user: admin
+    ipa_pass: topsecret
+
+- name: Create a 8 digit hotp for pinky with sha256 with specified validity times
+  community.general.ipa_otptoken:
+    uniqueid: Token123
+    enabled: true
+    otptype: hotp
+    digits: 8
+    secretkey: UMKSIER00zT2T2tWMUlTRmNlekRCbFQvWFBVZUh2dElHWGR6T3VUR3IzK2xjaFk9
+    algorithm: sha256
+    notbefore: 20180121182123
+    notafter: 20220121182123
+    owner: pinky
+    ipa_host: ipa.example.com
+    ipa_user: admin
+    ipa_pass: topsecret
+
+- name: Update Token123 to indicate a vendor, model, serial number (info only), and description
+  community.general.ipa_otptoken:
+    uniqueid: Token123
+    vendor: Acme
+    model: acme101
+    serial: SerialNumber1
+    description: Acme OTP device
+    ipa_host: ipa.example.com
+    ipa_user: admin
+    ipa_pass: topsecret
+
+- name: Disable Token123
+  community.general.ipa_otptoken:
+    uniqueid: Token123
+    enabled: false
+    ipa_host: ipa.example.com
+    ipa_user: admin
+    ipa_pass: topsecret
+
+- name: Rename Token123 to TokenABC and enable it
+  community.general.ipa_otptoken:
+    uniqueid: Token123
+    newuniqueid: TokenABC
+    enabled: true
+    ipa_host: ipa.example.com
+    ipa_user: admin
+    ipa_pass: topsecret
+'''
+
+RETURN = r'''
+otp:
```

Comment:
Here you claim your return value is called `otp`, while the module actually returns `otptoken`.

Reply:
Resolved in 59f47ff